src/gen_smtp_server_session.erl

```diff
@@ -709,7 +710,7 @@ handle_request({<<"STARTTLS">>, <<>>}, #state{socket = Socket, module = Module,
 			case ranch_ssl:handshake(Socket, [{packet, line}, {mode, list}, {ssl_imp, new} | TlsOpts2], 5000) of %XXX: see smtp_socket:?SSL_LISTEN_OPTIONS
 				{ok, NewSocket} ->
 					?log(debug, "SSL negotiation sucessful~n"),
-					ranch_ssl:setopts(NewSocket, [{packet, line}]),
+					ranch_ssl:setopts(NewSocket, [{packet, line}, binary]),
```

Comment:
for some reason, without this change `ssl` socket started to send lines as strings instead of binaries on OTP-24

Reply:
That is weird, sounds like a bug in OTP.

Reply:
Yes I traced this one down here: https://github.com/erlang/otp/issues/4585

Reply:
Ah, so we [fixed it for SMTP client](https://github.com/gen-smtp/gen_smtp/commit/f157a572d3ea11bc223c0e72c3e037532842c6c6), but forgot about the server :smile:
But good to know it's not some mystery but actual known OTP issue.